e2e/capl-cluster-flavors/kubeadm-full-capl-cluster/chainsaw-test.yaml

```diff
@@ -0,0 +1,216 @@
+              --config ${CLUSTERCTL_CONFIG:=${HOME}/.cluster-api/clusterctl.yaml} > kubeadm-full-cluster.yaml
+            check:
+              ($error == null): true
+    # Test for applying created yaml
+    - name: Apply generated cluster yaml
+      try:
+        - apply:
+            file: kubeadm-full-cluster.yaml
+        - assert:
+            file: assert-child-cluster-resources.yaml
+      catch:
+        - describe:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+            kind: LinodeMachine
+        - describe:
+            apiVersion: cluster.x-k8s.io/v1beta1
+            kind: Machine
+        - describe:
+            apiVersion: cluster.x-k8s.io/v1beta1
+            kind: MachineDeployment
+        - describe:
+            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+            kind: KubeadmControlPlane
+        - describe:
+            apiVersion: addons.cluster.x-k8s.io/v1alpha1
+            kind: HelmReleaseProxy
+        - describe:
+            apiVersion: cluster.x-k8s.io/v1beta1
+            kind: MachineSet
+
+    # Test to check if linodes are created
+    - name: Check if the linodes are created
+      try:
+        - script:
+            env:
+              - name: TARGET_API
+                value: api.linode.com
+              - name: TARGET_API_VERSION
+                value: v4beta
+              - name: URI
+                value: linode/instances
+              - name: FILTER
+                value: (to_string({"tags":($cluster)}))
+            content: |
+              set -e
+              curl -s \
+                -H "Authorization: Bearer $LINODE_TOKEN" \
+                -H "X-Filter: $FILTER" \
+                -H "Content-Type: application/json" \
+                "https://$TARGET_API/$TARGET_API_VERSION/$URI"
+            check:
+              ($error): ~
+              (json_parse($stdout)):
+                results: 2
+
+    # Get the KUBECONFIG of the child cluster for later use
+    - name: Get child cluster kubeconfig
+      try:
+        - script:
+            env:
+              - name: CLUSTER
+                value: ($cluster)
+              - name: NAMESPACE
+                value: ($namespace)
+              - name: CLUSTERCTL_CONFIG
+                value: (env('CLUSTERCTL_CONFIG'))
+            content: |
+              set -e
+              clusterctl get kubeconfig $CLUSTER -n $NAMESPACE > kubeadm-full-cluster-kubeconfig.yaml
+            check:
+              ($error == null): true
+
+    # Test for child cluster items: konnectivity
+    - clusters:
+        kubeadm-full-cluster:
+          kubeconfig: ./kubeadm-full-cluster-kubeconfig.yaml
+      name: Check child cluster resources
+      try:
+        - assert:
+            cluster: kubeadm-full-cluster
+            file: assert-child-cluster-deployments.yaml
+        - assert:
+            cluster: kubeadm-full-cluster
+            file: assert-child-cluster-daemonsets.yaml
```

Comment:
I would break out konnectivity stuff out of these files. It would be easier in future to figure out what exactly this test is testing and not have an ambiguity

Reply:
Sure, I can make that change. I was just testing all the daemon sets at once. But you're right, it will be less ambiguous and better to split them up.